cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -527,53 +533,78 @@ private void checkReconfigurability(NodeRef nodeRef, Pod pod, RestartContext res
         KafkaBrokerConfigurationDiff diff = null;
         KafkaBrokerLoggingConfigurationDiff loggingDiff = null;
         boolean needsReconfig = false;
-        // Always get the broker config. This request gets sent to that specific broker, so it's a proof that we can
-        // connect to the broker and that it's capable of responding.
-        if (!initAdminClient()) {
-            LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted, because it does not seem to responding to connection attempts", nodeRef);
-            reasonToRestartPod.add(RestartReason.POD_UNRESPONSIVE);
-            restartContext.needsRestart = false;
-            restartContext.needsReconfig = false;
-            restartContext.forceRestart = true;
-            restartContext.diff = null;
-            restartContext.logDiff = null;
-            return;
-        }
-        Config brokerConfig;
-        try {
-            brokerConfig = brokerConfig(nodeRef);
-        } catch (ForceableProblem e) {
-            if (restartContext.backOff.done()) {
+        if (nodeRef.controller()) {
+            LOGGER.debugCr(reconciliation, "Pod {} is a KRaft controller, checking if it requires a restart.", nodeRef);
+            if (kafkaAgentClient == null) {
+                kafkaAgentClient = initKafkaAgentClient();
+            }
+            Config controllerConfig = kafkaAgentClient.getNodeConfiguration(nodeRef.podName());
+            // This configuration property is used by the quorum check for rolling KRaft controllers
+            // and users are allowed to configure this property therefore we need to retrieve its current value.
+            Optional.ofNullable(controllerConfig.get(CONTROLLER_QUORUM_FETCH_TIMEOUT_MS_CONFIG_NAME)).ifPresent(config -> restartContext.controllerQuorumFetchTimeoutMs = Long.getLong(config.value()));
```

Comment:
Shouldn't it be the agent's responsibility to supply the default?

Reply:
If you are referring to the Kafka agent, how can it know the defaults? I mean, the agent is just returning the controller configuration from a file on the node, it doesn't have any knowledge of defaults for not specified values.

But thinking more about your point ... if we see the Kafka Agent as a "replacement" for the Kafka Admin (for this temporary workaround), maybe we should "hardcode" the defaults (got from documentation/Kafka codebase) in the agent itself? Should the agent return the content of the configuration file on the node + all other missing parameters (with corresponding defaults as values) as the roller was calling the Kafka Admin client?

Reply:
+1 to what @ppatierno said. Currently we have no way of knowing the default values from KafkaAgent. In KafkaRoller, we could make describeConfigs() call to normal brokers' admin API (we already do that for quorum check), to figure out the defaults but that would be an additional call to the brokers.

Reply:
OK, let's keep this as it is for now.